lib/server/monalisaSynchronizers/AbstractHTTPSynchronizer.js

```diff
@@ -0,0 +1,88 @@
+/**
+ *
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { Log } = require('@aliceo2/web-ui');
+
+/**
+ * AbstractHTTPSynchronizer
+ * The class provides schema for excecuting process of data synchronization with external service
+ * Its behaviour can be customized with overriding abstract methods
+ * Communication is carried via http with json data format
+ */
+class AbstractHTTPSynchronizer {
```

Comment:
I don't understand the point of this abstraction, maybe it would make more sense if I see an implementation.
What is the advantage of having a class that does all of this?
If I rephrased, what is the purpose of this class? It has a lot of reason to change (if the API changes, to gRPC for example, if we change the data storage and so on).

Reply:
There is need to fetch data from three endpoints of MonALISA. Each process of data fetching looks more or less the same P: (fetch data) -> (adjust data) -> (update DB).
- P1 - fetching data passes
- P2 - fetching data passes details - need to be done alongside with P1
- P3 - fetching Monte Carlo productions (Simulation passes)

Due to this there is an abstract class which will be inherited by three other.
-----
If API change and we have no such a abstraction I think there will be more changes required.
Storage? What about Sequelize?

Reply:
For P1 and P2 it is like
for each DB update of P1 do entire P2 within one transaction - what is to update Data Pass and its details together in order to keep data consistent

Reply:
Please put at least one concrete class first, then when you bring the second you may bring the abstraction.
Here, I see no really advantages of this abstraction except a lot of configurable code that will always be really specific.
Please, put the first concrete class in a first PR, then the abstraction with a second concrete class when it makes sense. Or, if you use the abstraction somewhere independently of the concrete classes I'm also ok.

Reply:
Here you are mixing two low level concepts: database, and API. If you want to have such a thing, it should be very generic, for example `ExternalDataSynchronizer`, to which you provide a way to fetch data from API, and a way to write data to storage, with adapters on both sides... But it's probably too much work for nothing.